app/controllers/api/v1/accounts/relationships_controller.rb

```diff
@@ -5,7 +5,7 @@ class Api::V1::Accounts::RelationshipsController < Api::BaseController
   before_action :require_user!
 
   def index
-    accounts = Account.where(id: account_ids).select('id')
+    accounts = Account.without_suspended.where(id: account_ids).select('id')
```

Comment:
Hm, maybe this should keep suspended accounts, so one can unfollow/block/etc. suspended accounts while those are suspended, in case they get reinstated.

Reply:
I'm going for consistency with previous behaviour. Previous behaviour where account data was deleted immediately meant that all following/muting/blocking/etc relationships were erased the moment the account was suspended. And unless the suspension is cancelled within 30 days, all of it will still be deleted in the end. So it makes sense for me to keep it that way. If an account is unsuspended, *then* other users may worry about following/unfollowing/muting/unmuting etc.

Reply:
Yes, but we're precisely changing the behavior here, to have suspension not be necessarily definitive. Regardless of the 30 days thing (which isn't implemented yet, and, depending on how federation is handled, might not be handled the same way by remote instances).

Reply:
No, it is implemented in the other PR.

If another server does not handle suspension the same way, then the account will not look suspended *there* and *their* users will be able to continue to unfollow, mute, and block without issue. Then after 30 days or earlier, the Delete will arrive there.

Reply:
I meant, if another implementation federates the suspension info but doesn't implement the 30 days-to-delete thing (either having a longer delay or no automatic deletion at all), there is no bound between the time an account is suspended and the time an account is reinstated. This can be very surprising if you have a “suspended” follower not listed anymore and they reappear months ago from seemingly nowhere.

Reply:
I fear that by including suspended users on followers lists and allowing following/unfollowing them it takes away from the seriousness of a suspension in the eyes of the users. Imagine "Why is \@Hitler8814 still in my followers, I have to block them myself?! This platform does nothing!"

You bring up a good point, what if another (malicious) server marks an account as suspended but allows that account to operate as normal, in that case the lacking controls would become a liability. But I propose that we move that discussion to federation of suspensions. While suspensions do not federate, a suspended account is merely a normal account on other servers up until it is deleted as usual.

Reply:
I guess *following* shouldn't be allowed, but muting, blocking, unfollowing should.

Good point about an obviously offending account still appearing in followers. I was more viewing it from the perspective ofs someone with a locked account, and considering an account that isn't *obviously* offensive but has been suspended for reasons I may or may not be aware of. I am unsure what the exact solution for this should be, but I'm still uncomfortable with the following relationship being hidden from me, the person being followed or following them.